File: 242-ValidAnagram/242-ValidAnagram.cpp
// Last updated: 16/10/2025, 04:27:37
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()) return false;
        unordered_map<char,int> mp;
        for(int i=0;i<s.size();i++){
            mp[s[i]]++;
        }
        for(int i=0;i<t.size();i++){
            mp[t[i]]--;
            if(mp[t[i]] == 0) mp.erase(t[i]);
        }
        
        if(mp.empty()) return true;
        else return false;
    }
};

File: 242-ValidAnagram/242-ValidAnagram.java
// Last updated: 16/10/2025, 04:27:32
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s == null || t == null || s.length() != t.length()) return false;
        char[] sChars = s.toCharArray();
        char[] tChars = t.toCharArray();
        HashMap<Character, Integer> map = new HashMap<>();
        for (char c : sChars
             ) {
            if (map.containsKey(c)) {
                map.put(c, map.get(c) + 1);
            } else {
                map.put(c, 1);
            }
        }
        for (char c : tChars
             ) {
            if (map.containsKey(c) && map.get(c) > 0) {
                map.put(c, map.get(c) - 1);
                if(map.get(c) == 0){
                    map.remove(c);
                }
            }
        }
        return map.isEmpty() ? true : false;
    }
}